Source/WebCore/css/CSSRule.cpp

```diff
@@ -70,6 +72,18 @@ bool CSSRule::hasStyleRuleAncestor() const
     return false;
 }
 
+CSSParserEnum::IsNestedContext CSSRule::nestedContext() const
+{
+    for (RefPtr parentRule = this->parentRule(); parentRule; parentRule = parentRule->parentRule()) {
+        if (is<CSSStyleRule>(parentRule))
+            return CSSParserEnum::AncestorRuleType::Style;
+        if (is<CSSScopeRule>(parentRule))
+            return CSSParserEnum::AncestorRuleType::Scope;
+    }
+
+    return { };
+}
```

Comment:
Traversing ancestors is not great but I suppose it is not a practical problem.

Reply:
It's only happening on CSSOM indeed